src/main/java/net/dries007/tfc/common/container/RunnableSlot.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed under the EUPL, Version 1.2.
+ * You may obtain a copy of the Licence at:
+ * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ */
+
+package net.dries007.tfc.common.container;
+
+import javax.annotation.Nonnull;
+
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.item.ItemStack;
+import net.minecraftforge.items.IItemHandler;
+import net.minecraftforge.items.SlotItemHandler;
+
+public class RunnableSlot extends SlotItemHandler
```

Comment:
This is unnecessary, the functionality exists in `CallbackSlot` already. Implement `ISlotCallback` on the container and override the methods `onTake` and `isItemValid`.

Also minding the fact that `RunnableSlot` is a horrible name in of itself.